### PR TITLE
Add build rules for stratum_bcm tar package

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR creates Bazel infrastructure to build a `stratum_bcm` package.
On its own it does not do much, but it can help to create Docker images and the like by bundling up all the required files.

Progress towards #94

### CODE REVIEW

File: stratum/hal/bin/bcm/standalone/BUILD [hunk 1]
```diff
@@ -88,3 +90,17 @@ stratum_package(
...
+    extension = "tar.gz",
+    srcs = [
+        ":stratum_bcm",
+        "@com_github_opennetworkinglab_sdklt//:kernel_modules",
+        "@com_github_opennetworkinglab_sdklt//:sdklt_cli",
+    ],
+    deps = [
+        "//stratum/hal/config:configs",
+    ],
+    mode = "0755",
```
Comment (Yi-Tseng) [inline stratum/hal/bin/bcm/standalone/BUILD]: This also overrides mode of all configs. Is it possibe to keep 644 for all configs?

Comment (pudelkoM) [reply stratum/hal/bin/bcm/standalone/BUILD]: yes. How about this:
```bash
tar --list -v -f bazel-bin/stratum/hal/bin/bcm/standalone/stratum_bcm_package.tar.gz
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./
-rwxr-xr-x 0/0        56735344 1970-01-01 00:00 ./stratum_bcm
-rwxr-xr-x 0/0           52808 1970-01-01 00:00 ./linux_ngbde.ko
-rwxr-xr-x 0/0          188880 1970-01-01 00:00 ./linux_ngknet.ko
-rwxr-xr-x 0/0         8700920 1970-01-01 00:00 ./sdklt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/
-rw-r--r-- 0/0            1647 1970-01-01 00:00 ./stratum_configs/bcm_hardware_specs.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-accton-as7712-32x-r0/
-rw-r--r-- 0/0           47184 1970-01-01 00:00 ./stratum_configs/x86-64-accton-as7712-32x-r0/SDKLT.yml
-rw-r--r-- 0/0          107683 1970-01-01 00:00 ./stratum_configs/x86-64-accton-as7712-32x-r0/base_bcm_chassis_map.pb.txt
-rw-r--r-- 0/0            3395 1970-01-01 00:00 ./stratum_configs/x86-64-accton-as7712-32x-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            3173 1970-01-01 00:00 ./stratum_configs/x86-64-accton-as7712-32x-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-32x-r0/
-rw-r--r-- 0/0            5298 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-32x-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            3328 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-32x-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-65x-r0/
-rw-r--r-- 0/0           10610 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-65x-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            5732 1970-01-01 00:00 ./stratum_configs/x86-64-accton-wedge100bf-65x-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-dell-z9100-c2538-r0/
-rw-r--r-- 0/0           44011 1970-01-01 00:00 ./stratum_configs/x86-64-dell-z9100-c2538-r0/base_bcm_chassis_map.pb.txt
-rw-r--r-- 0/0            3395 1970-01-01 00:00 ./stratum_configs/x86-64-dell-z9100-c2538-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            3328 1970-01-01 00:00 ./stratum_configs/x86-64-dell-z9100-c2538-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-delta-ag9064v1-r0/
-rw-r--r-- 0/0           10646 1970-01-01 00:00 ./stratum_configs/x86-64-delta-ag9064v1-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            5842 1970-01-01 00:00 ./stratum_configs/x86-64-delta-ag9064v1-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10056-r0/
-rw-r--r-- 0/0            9076 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10056-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            5121 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10056-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10064-r0/
-rw-r--r-- 0/0           10594 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10064-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            5707 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d10064-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d5254-r0/
-rw-r--r-- 0/0            8755 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d5254-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            4961 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d5254-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d7032q28b-r0/
-rw-r--r-- 0/0           55638 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d7032q28b-r0/SDKLT.yml
-rw-r--r-- 0/0           19059 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d7032q28b-r0/base_bcm_chassis_map.pb.txt
-rw-r--r-- 0/0            3380 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d7032q28b-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            3065 1970-01-01 00:00 ./stratum_configs/x86-64-inventec-d7032q28b-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-quanta-ix1-rangeley-r0/
-rw-r--r-- 0/0           44209 1970-01-01 00:00 ./stratum_configs/x86-64-quanta-ix1-rangeley-r0/SDKLT.yml
-rw-r--r-- 0/0           19116 1970-01-01 00:00 ./stratum_configs/x86-64-quanta-ix1-rangeley-r0/base_bcm_chassis_map.pb.txt
-rw-r--r-- 0/0            3386 1970-01-01 00:00 ./stratum_configs/x86-64-quanta-ix1-rangeley-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            3638 1970-01-01 00:00 ./stratum_configs/x86-64-quanta-ix1-rangeley-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf2556x-1t-r0/
-rw-r--r-- 0/0            9718 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf2556x-1t-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            4967 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf2556x-1t-r0/phal_config.pb.txt
drwxr-xr-x 0/0               0 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf6064x-t-r0/
-rw-r--r-- 0/0           10594 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf6064x-t-r0/chassis_config.pb.txt
-rw-r--r-- 0/0            5842 1970-01-01 00:00 ./stratum_configs/x86-64-stordis-bf6064x-t-r0/phal_config.pb.txt
```

Comment (Yi-Tseng) [reply stratum/hal/bin/bcm/standalone/BUILD]: lgtm

**Comment by ccascone** [comment]
As discussed f2f, I think that publishing tarballs as a GitHub release for every commit in master is probably a bad idea. Eventually, Stratum will have a proper release process in place which will conflict with this. Alternative options you might consider are:
- Circle CI ability to [store artifacts](https://circleci.com/docs/2.0/artifacts/)
- Use a public S3 bucket in the AWS ONF account
- Distribute Docker images instead of tarballs, as we do for `stratum_bf, and upload to Docker Hub

**Comment by pudelkoM** [comment]
> As discussed f2f, I think that publishing tarballs as a GitHub release for every commit in master is probably a bad idea. Eventually, Stratum will have a proper release process in place which will conflict with this. Alternative options you might consider are:
> 
>     * Circle CI ability to [store artifacts](https://circleci.com/docs/2.0/artifacts/)
> 
>     * Use a public S3 bucket in the AWS ONF account
> 
>     * Distribute Docker images instead of tarballs, as we do for `stratum_bf`, and upload to Docker Hub

The main motivation comes from making it as fast & easy (<5 min) as possible to get a Stratum running on a Broadcom switch. Also see PR #88.

To me CircleCI artifacts become irrelevant with the first step already:
> 1. Create a personal API token and copy it to a clipboard.
Random users don't have a CircleCI account.

**Comment by pudelkoM** [comment]
> To me CircleCI artifacts become irrelevant with the first step already:

I have to take that back. For simple, single file downloads no API key is required and CircleCI artifacts serve our purpose very well. Even more, it allows creating artifacts for development branches as well.

Only thing to keep in mind:
> Artifacts are designed to be useful around the time of the build. It is best practice not to rely on artifacts as a software distribution mechanism with long term future guarantees.

But that's ok for now.

Thanks for the hints @ccascone  and sorry for the notification spam!

**Pull request merged**